products-controller.js

```diff
@@ -14,8 +14,24 @@ let products = [{
 ]
 
 // TODO -> end points handlers:
-
+const getByCategory = (req, res)=>{
+    let productsByCategory = []
+    const category = req.params.category;
+    products.forEach((i)=>{
+        if(i.category===category){
+            productsByCategory.push(i);
+        }
+    })
+    if(productsByCategory.length==0){
+        res.status(404).send(`Error: cateogry '${category}' not found`);
+    }
```

Comment:
When the expected answer is an array not a specific element, it's more natural to respond with an empty array than an error, since nothing actually went wrong, the user asked for all our products under a certain category and we don't have any.

We return NOTFOUND error when the user is asking for a product with a given id but that id doesn't exist, that means either the user is trying to access a deleted or non-existent product or our system is losing data somehow and it's probably the former than the laster.